updated 3D noise array function

Julian-Wyatt opened the pull request:
given array coordinate inputs x,y,z, the function returns 3D array of shape (z,y,x) instead of 1D array.

lmas commented:
Hi thank you for this. Would you be willing to update it further and match it with #25?
I'm asking you because your PR was smaller of the two.

In short:
- Add docstring and type hints to to the public func noise3array (not _noise3array)
- Add back the old input variable names and use x_i for the loop vars

lmas commented on opensimplex/opensimplex.py:
```diff
@@ -107,11 +107,13 @@ def _noise2a(x, y, perm):
+    for z in prange(Z.size):
+        for y in prange(Y.size):
+            for x in prange(X.size):
+                noise[(y * Y.size + x)+(z * Y.size * X.size)] = _noise3(X[x], Y[y], Z[z], perm, perm_grad_index3)
```
Mixing upper/lower case single variables is just confusing and hard to read.
It's better to keep the original inputs (lower case x, y, z) so the function signature doesn't change unnecessarily and instead call the loop vars something else (ie, add some extra letter).

Julian-Wyatt replied:
Sure no worries! 
Shall get onto that now

Julian-Wyatt commented:
Hope those two commits fix the issues and help. I've used the other PR for consistency.

lmas commented:
Whoa hold up now mister. I appreciate your time and effort, thanks, but now you've started to add in _partial code_ that's unrelated to your initial PR.

_It's not my intention to sound like an ungrateful muppet, I just want to help you get it right while keeping things clean for everyone else :)_

It's way easier to work with small and clean PRs that contains just a single feature/update/whatever. So your initial PR was good. But it would had been better to open up a new PR for that third commit instead.

lmas commented:
Can you revert the commit with the new fractal noise func and add it as a separate PR? And please run the new formatter on your code.

Julian-Wyatt commented:
I've removed the ident from the doc string.
I've reverted the final commit, adding the potential new feature. Instead added this as another PR, building on from this branch.

lmas commented:
Thank you, looking better now! I don't have time during the week so I'll try to verify and merge this weekend 👍
